samples/auth-user-sample/authUserFlow.js

```
@@ -0,0 +1,155 @@
+  const authorizeUrl = oAuth2Client.generateAuthUrl({
+    access_type: 'offline',
+    scope: 'https://www.googleapis.com/auth/bigquery',
+    prompt: 'consent',
+  });
+
+  console.info(
+    `Please visit this URL to authorize this application: ${authorizeUrl}`
+  );
+
+  const code = await rl.questionAsync('Enter the authorization code: ');
+  const tokens = await main.exchangeCode(code);
+  rl.close();
+
+  return tokens;
+}
+
+// Exchange an authorization code for an access token
+async function exchangeCode(code) {
+  const oAuth2Client = new OAuth2Client(
+    keys.installed.client_id,
+    keys.installed.client_secret,
+    keys.installed.redirect_uris[0]
+  );
+
+  const r = await oAuth2Client.getToken(code);
+  console.info(r.tokens);
+  return r.tokens;
+}
+
+async function authFlow(projectId = 'project_id') {
+  /**
+   * TODO(developer):
+   * Save Project ID as environment variable PROJECT_ID="project_id"
+   * Uncomment the following line before running the sample.
+   */
+  // projectId = process.env.PROJECT_ID;
+
+  const tokens = await main.getRedirectUrl();
+
+  const credentials = {
+    type: 'authorized_user',
+    client_id: keys.installed.client_id,
+    client_secret: keys.installed.client_secret,
+    refresh_token: tokens.refresh_token,
+  };
+
+  return {
+    projectId,
+    credentials,
+  };
+}
+// [END bigquery_auth_user_flow]
+async function query() {
+  const {BigQuery} = require('@google-cloud/bigquery');
+
+  const credentials = await main.authFlow();
+  const bigquery = new BigQuery(credentials);
+
+  // Queries the U.S. given names dataset for the state of Texas.
+  const query = ` SELECT name, SUM(number) as total
+  FROM \`bigquery-public-data.usa_names.usa_1910_current\`
+  WHERE name = 'William'
+  GROUP BY name;`;
+
+  // For all options, see https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query
+  const options = {
+    query: query,
+  };
+
+  // Run the query as a job
+  try {
```

No need to for the try catch here.  Just let it throw!